scipy/optimize/_optimize.py

```diff
@@ -271,6 +271,12 @@ class OptimizeWarning(UserWarning):
     pass
 
 
+def _check_c1_c2(c1, c2):
+    if not (0 < c1 < c2 < 1):
+        raise ValueError("'c1' and 'c2' do not satisfy"
+                         "'0 < c1 < c2 < 1'.")
```

Comment:
Could this not be moved into `_line_search_wolfe12`?

Reply:
I would rather add it in `line_search_wolfe1` and/or `line_search_wolfe2` because `scipy.optimize.line_search` only calls [line_search_wolfe2](https://github.com/scipy/scipy/blob/686422c4f0a71be1b4258309590fd3e9de102e18/scipy/optimize/_linesearch.py#L183-L322). Hence, if `c1` and `c2` are not checked there, there will be an unconsistency in how inputs are checked. Moreover, it seems like `_line_search_wolfe12` is only used here, while `line_search_wolfe1` and/or `line_search_wolfe2` could be used by any future algorithm requiring and exact line_search.

Reply:
Good point, let's add it to the underlying functions.